lib/data.php

```diff
@@ -50,6 +51,37 @@ function reportExists($reportId, $reportsArray) {
    return false;
 }
 
+/**
+ * Get fields for a given report.
+ * @param Integer $reportId The report id
+ * @return Array A list of associative arrays, each with keys: field_name and field_label.
+ */
+function getReportFields($reportId) {
```

Comment:
This function was extracted - no logic changed. Initially I was going to compute the `bucketByExistsOnReport` boolean here in the controller and pass it to `SummaryUIProcessor::summaryConfig($bucketByExistsOnReport)`, but realized the logic could all be moved into `SummaryUIProcessor`.

I could probably move this function in there as well with a bit of refactoring. Maybe I'll create an issue since it'll be a bit of work.

Reply:
If there are advantages to refactoring to move logic to `SummaryUIProcessor`, I'd tackle it right away. That way it's fresh in your mind. Also, from my experience this kind of thing just doesn't get done if it's left on the backlog.

Reply:
Good point, I'll go ahead and do it in a second PR.